### DISCUSSION TITLE
After submitting form get error of page expired 419 in Livewire

### DISCUSSION THREAD

**Discussion by qalildarwish** [livewire/livewire#7012] [1. Help] [answered]
**I have Livewire component which has modal with form for the first time submitting it works correctly after first attempt it gave page expired error!**
My component is below
**1:index.blade.php**  the blade file 
<div>
<div class="col-sm-12">
        <div class="card">
            <div class="card-header d-flex justify-content-between flex-wrap">
                <div class="header-title">
                    <h4 class="card-title mb-0">Branches List</h4>
                </div>
                <div class=""> 
                    <a href="#" class=" text-center btn btn-primary btn-icon mt-lg-0 mt-md-0 mt-3" data-bs-toggle="modal" data-bs- 
                           target="#staticBackdrop-1">
                        <i class="btn-inner">
                            <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                            <circle cx="12" cy="12" r="10"></circle><line x1="12" y1="8" x2="12" y2="16"></line><line x1="8" y1="12" x2="16" 
                           y2="12"></line>

                            </svg>
                        </i>
                        <span>New Branch</span>
                    </a>
            </div>
    </div>

     
    <div wire:ignore.self class="modal fade" id="staticBackdrop-1" data-bs-backdrop="static" data-bs-keyboard="false" tabindex="-1" aria- 
     labelledby="staticBackdropLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h5 class="modal-title" id="modelTitle">New Branch</h5>
                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
            </div>
            <div class="modal-body">
                 
            
            
                 <form wire:submit.prevent="storeBranch">      
                 <div class="form-group">
                    <label for="email" class="form-label">Branch Name</label>
                    <input type="text" class="form-control" wire:model.defer="branch_name" autofocus autocomplete="off">
                    @error ('branch_name')<small class="text-danger">{{$message}}</small> @enderror 
                </div>
               
                <div class="form-group">
                    <label for="email" class="form-label">Owner Name</label>
                    <input type="text" class="form-control"  wire:model.defer="owner">
                    @error ('owner')<small class="text-danger">{{$message}}</small> @enderror
                </div>
                <div class="form-group">
                    <label for="email" class="form-label">JawazNo</label>
                    <input type="text" class="form-control"  wire:model.defer="jawazNo" id="jawazNo">
                    @error ('jawazNo')<small class="text-danger">{{$message}}</small> @enderror
                </div>

                <div class="form-group">
                    <label for="email" class="form-label">Jawaz Type</label>
                    <input type="text" class="form-control"   wire:model.defer="jawazType" id="jawazType">
                    @error ('jawazType')<small class="text-danger">{{$message}}</small> @enderror
                </div>
                
           
                
                <div class="form-group ">
                    <label for="email" class="form-label">Address</label>
                    <input type="text" class="form-control" wire:model.defer="address" id="address">
                    @error ('address')<small class="text-danger">{{$message}}</small> @enderror
                </div>

                <div class="form-group">
                    <label for="email" class="form-label">Phone</label>
                    <input type="text" class="form-control"   wire:model.defer="phone" id="phone">
                    @error ('phone')<small class="text-danger">{{$message}}</small> @enderror
                </div>

                <div class="text-start mt-2">
                    <button type="button" class="btn btn-primary" wire:click="storeBranch"> Save</button>
                    <button type="button" class="btn btn-danger" data-bs-dismiss="modal">Cancel</button>
                </div>
            
                </form>
            </div>
            
        </div>
    </div>
</div>

    </div>
</div>
</div>

[[@push('script')
<script>
    window.addEventListener('closeModal', event => {
        $("#staticBackdrop-1").modal('hide');       

        });
</script>

  <script>
            window.addEventListener('swal', function(e){
                Swal.fire({
                title: '<strong>Success!</strong>',
                text:'Record has been successfully inserted!',
                icon: 'success',
                timer:'2000'
                })
          });

 </script>
   @endpush
](url)
**2:index.php** Server side 
    public $branch_name,$owner,$jawazNo,$jawazType,$address,$phone;
    public function rules(){
     return [
            
         'branch_name' => 'required|regex:/^[\pL\s\-]+$/u|unique:branches',
         'owner' => 'required|regex:/^[\pL\s\-]+$/u',
         'jawazType' => 'required',
         'jawazNo' => 'required',
         'address' => 'required|string|max:255',
         'phone' => 'required|string|max:255',
 
     ];
 
    }
    public function resetInput(){
     $this->branch_name=NULL;
     $this->owner=NULL;
     $this->jawazType=NULL;
     $this->jawazNo=NULL;
     $this->address=NULL;
     $this->phone=NULL;
    }

     
  

      public function storeBranch(){
      $validateDate=$this->validate();
 
         $Branch= Branch::create([
               'branch_name'=>$this->branch_name,
               'owner'=>$this->owner,
               'jawazType'=>$this->jawazType,
               'jawazNo'=>$this->jawazNo,
               'address'=>$this->address,
               'phone'=>$this->phone
           ]);
           
           session()->flush('success','Branch has been successfully added!');
            $this->dispatch('closeModal'); 
            $this->resetInput();

           $this->dispatch('swal',[
            'type'=>'success',
            'title'=>'Okay',
            'text'=>'Record has been successfully inserted!',
            'icon'=>'success',
            'timer'=>3000

        ]);

      }
 
    public function render()
    {
        $branches=Branch::all();
        return view('livewire.branch.index',compact('branches'))
        ->extends('layout')
        ->section('content');
    }

**Comment by shokanshi** [comment, marked as answer]
419 is error returned when CSRF token has expired.

You can update the following code in `app\Exceptions\Handler.php`

```
    public function register()
    {
        $this->renderable(function (\Exception $e) {
            if ($e->getPrevious() instanceof \Illuminate\Session\TokenMismatchException) {
                session()->flash('status', __('Session has expired. Please login again.'));
                return redirect()->route('login');
            };
        });

        $this->reportable(function (Throwable $e) {
            //
        });
    }
```
When the token has expired, user will be redirected to the login page.
